src/Airac.cpp

```diff
@@ -107,69 +107,41 @@ void Airac::readAirways(const QString& directory) {
             continue;
 
         QString id = list[0];
-        double lat = list[1].toDouble(&ok);
-        if(!ok) {
-            qWarning() << "Airac::readAirways() unable to parse lat (double):" << list;
-            continue;
-        }
-        double lon = list[2].toDouble(&ok);
-        if(!ok) {
-            qWarning() << "Airac::readAirways() unable to parse lon (double):" << list;
-            continue;
-        }
-
-        Waypoint *start = waypoint(id, lat, lon, 1);
-        if(start == 0) {
-            start = new Waypoint(id, lat, lon);
-            waypoints[start->label].insert(start);
-        }
+        QString regionCode = list[1];
+        Waypoint *start = waypoint(id, regionCode);
 
         id = list[3];
-        lat = list[4].toDouble(&ok);
-        if(!ok) {
-            qWarning() << "Airac::readAirways() unable to parse lat (double):" << list;
-            continue;
-        }
-        lon = list[5].toDouble(&ok);
-        if(!ok) {
-            qWarning() << "Airac::readAirways() unable to parse lon (double):" << list;
-            continue;
-        }
-
-        Waypoint *end = waypoint(id, lat, lon, 1);
-        if(end == 0) {
-            end = new Waypoint(id, lat, lon);
-            waypoints[end->label].insert(end);
-        }
+        regionCode = list[4];
+        Waypoint *end = waypoint(id, regionCode);
```

Comment:
the result could be a null ptr here.
```suggestion
        Waypoint *end = waypoint(id, regionCode);
        if(end == 0) {
            end = new Waypoint(id, lat, lon);
            waypoints[end->label].insert(end);
        }
```

Reply:
I'm not sure where you get the lat and lon values? The airway segments are purely defined via the waypoint name and region code, so I'm not sure your suggestion would work here.

We should of course check for the nullptr, but IMO if it can't find the waypoint, we should fail.

Reply:
You are right. I forgot that this changed there and have only compared with the current control flow there. 
Anyways, make sure you don't continue with an `end == 0`. I believe that's where the crashes come from. There is a core dump when that happens but I haven't looked into it.